enhancements/sig-architecture/32-extensiblescheduling/32-extensiblescheduling.md

```diff
@@ -0,0 +1,363 @@
+# placement extensible scheduling
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/website/)
+
+## Summary
+
+The proposed work provides an API to represent a scalar value of the managed cluster, to support placement extensible scheduling.
+
+## Motivation
+
+When implementing placement resource based scheduling, we find in some cases, prioritizer needs extra data (more than the defualt value provided by `ManagedCluster`) to calculate the score of the managed cluster. For example, there is requirement to schedule based on resource monitoring data from the cluster.
+
+So we want a more extensible way to support scheduling based on customized value.
+
+### Goals
+- Design a new API(CRD) to contain a scalar value for each managed cluster.
+- Let placement prioritizer maintain the lifecycle (creating/deleting) of the CR.
+- Let placement prioritizer support rating clusters with the scalar value of the CR.
+
+### Non-Goals
+- How a customized agent update the scalar value into the CR.
+- Placement filter will not support the new API(CRD).
+
+## Proposal
+
+### User Stories
+
+#### Story 1: Users could use the data pushed from each managed cluster to select clusters.
+  - On each managed cluster, there is a customized agent monitoring the resource usage (eg. CPU ratio) of the cluster. It will calculate a scalar value and push the result to hub.
+  - As an end user, I can configure placement yaml to use this scalar value to select clusters.
+  
+#### Story 2: Users could use the metrics collected on hub to select clusters.
+  - On the hub, there is a customized agent to get metrics (eg. cluster allocatable memory) from Thanos. It will generate a scalar value for each cluster.
+  - As an end user, I can configure placement yaml to use this scalar value to select clusters.
+
+#### Story 3: Disaster recovery workload could be automatically switched to an avaliable cluster.
+  - A user has two clusters, a primary and backup cluster on which workload storage is configured so storage are synced from primary cluster to backup.
+  - As an end user, I want to deploy workloads on the primary cluster first. And when primary cluster is down, the workload should be automatically switched to the backup cluster.
+
+#### Story 4: Users could use the data provided by third party controllers to select clusters.
+  - A third party controller can comprehensively evaluate cluster on latency, region, iops etc. and rate cluster according to the SLA the cluster can provide.
+  - As an end user, I can configure placement yaml to use this scalar value to select clusters.
+
```

Comment:
For the story, maybe `sort clusters` is better than `select clusters`, as for the current OCM usage, there will some clusters in a placement, what we will do is to sort these clsuters with reported values, and end users can select the top one or top N from the placementdecision. For `select clusters`, the clusters in the placementdecision need to be updated (add/delete) with the reported values, the basic behavior is changed, and the performance may be impacted.

Reply:
@qiujian16 @elgnay any comments for this?

Reply:
We don't support `sort clusters` in placement decision is because it will trigger a frequent update. For example, when cluster score (high to low) changes from "cluster1, cluster2, cluster3" to "cluster3, cluster2, cluster1", `sort clusters` will trigger an update, while if use `select clusters`, it keeps "cluster1, cluster2, cluster3".

Reply:
Only show the top 1 cluster has the same issue, all the clusters need to be sorted when the cluster scores are changed.